### ISSUE TITLE
Custom landing zone variable validation only checks first key

### ISSUE THREAD

**Issue by larsakerlund** [Azure/terraform-azurerm-caf-enterprise-scale#61]
The validation of the `custom_landing_zone` keys only checks the value of the first key. This makes it possible to set invalid key names (which fails later).

Example inputs:
```terraform
# Passes validation (and fails during archetype variable validation)
custom_landing_zones = {
  (var.root_id) = {
    display_name               = var.root_name
    parent_management_group_id = data.azurerm_client_config.current.tenant_id
    subscription_ids           = []
    archetype_config = {
      archetype_id   = "es_root"
      parameters     = {}
      access_control = {}
    }
  }

  test_key = {
    display_name               = "Test"
    parent_management_group_id = var.root_id
    subscription_ids           = []
    archetype_config = {
      archetype_id   = "es_landing_zones"
      parameters     = {}
      access_control = {}
    }
  }
}

# Does not pass validation 
custom_landing_zones = {
  test_root = {
    display_name               = var.root_name
    parent_management_group_id = data.azurerm_client_config.current.tenant_id
    subscription_ids           = []
    archetype_config = {
      archetype_id   = "es_root"
      parameters     = {}
      access_control = {}
    }
  }
}
```

**Comment by krowlandson** [comment]
Great catch... I had forgotten to go back to update this logic.

We're reviewing this PR in conjunction with PR #59 so I would appreciate your response to my question on the related issue #58 before we merge these.

Thank you

**Comment by krowlandson** [comment]
Closing as fixed by #62
